filemutex_flock.go

```diff
@@ -57,3 +58,15 @@ func (m *FileMutex) RUnlock() {
 	}
 	m.mu.RUnlock()
 }
+
+// Close does an Unlock() combined with closing and unlinking the associated
+// lock file. You should create a New() FileMutex for every Lock() attempt if
+// using Close().
+func (m *FileMutex) Close() {
+	if err := syscall.Flock(m.fd, syscall.LOCK_UN); err != nil {
+		panic(err)
+	}
+	syscall.Close(m.fd)
+	syscall.Unlink(m.path)
+	m.mu.Unlock()
```

Comment:
The case I'm thinking of is where I have a FileMutex that I repeatedly Lock and Unlock, and then at the end of all this I want to Close it. This means the mutex would be in the unlocked state when I want to close it, and it would feel weird to be required to lock it before closing it.